common/thorhelper/roxiehelper.cpp

```diff
@@ -1598,15 +1598,21 @@ void CSafeSocket::setHttpMode(const char *queryName, bool arrayMode, HttpHelper
     mlResponseFmt = httphelper.queryResponseMlFormat();
     respCompression = httphelper.getRespCompression();
     heartbeat = false;
-    assertex(contentHead.length()==0 && contentTail.length()==0);
+
+    //reset persistent http connection
+    ForEachItemIn(idx, queued)
+        free(queued.item(idx));
+    queued.kill();
+    lengths.kill();
+
     if (mlResponseFmt==MarkupFmt_JSON)
     {
         contentHead.set("{");
         contentTail.set("}");
     }
     else
     {
-        StringAttrBuilder headText(contentHead), tailText(contentTail);
+        StringAttrBuilder headText(contentHead.clear()), tailText(contentTail.clear());
```

Comment:
If I see well, the contentHead.clear() has no return value (void) and a StringAttrBuilder object needs a reference. 
I think the proper way is:
contentHead.clear();
contentTail.clear();
StringAttrBuilder headText(contentHead), tailText(contentTail);
(And it is build, I tested.)